src/main/java/org/springframework/data/redis/cache/RedisCacheWriter.java

```diff
@@ -104,29 +106,43 @@ static RedisCacheWriter lockingRedisCacheWriter(RedisConnectionFactory connectio
 
 		Assert.notNull(connectionFactory, "ConnectionFactory must not be null");
 
-		return new DefaultRedisCacheWriter(connectionFactory, sleepTime, lockTtlFunction, CacheStatisticsCollector.none(),
-				batchStrategy);
+		return new DefaultRedisCacheWriter(connectionFactory, sleepTime, lockTtlFunction,
+				CacheStatisticsCollector.none(), batchStrategy);
 	}
 
 	/**
-	 * Write the given key/value pair to Redis and set the expiration time if defined.
+	 * Get the binary value representation from Redis stored for the given key.
 	 *
-	 * @param name The cache name must not be {@literal null}.
-	 * @param key The key for the cache entry. Must not be {@literal null}.
-	 * @param value The value stored for the key. Must not be {@literal null}.
-	 * @param ttl Optional expiration time. Can be {@literal null}.
+	 * @param name must not be {@literal null}.
+	 * @param key must not be {@literal null}.
+	 * @return {@literal null} if key does not exist.
+	 * @see #get(String, byte[], Duration)
 	 */
-	void put(String name, byte[] key, byte[] value, @Nullable Duration ttl);
+	@Nullable
+	default byte[] get(String name, byte[] key) {
```

Comment:
Let's do the defaulting the other way round, that we introduce `get(String name, byte[] key, @Nullable Duration ttl)` as default method to not break existing implementations.

Reply:
Good point.